cogs/music.py

```diff
@@ -0,0 +1,553 @@
+    async def music_status(self, ctx):
+        """Displays current status of the bot"""
+
+        status_embed = discord.Embed(colour=random.randint(0, 0xFFFFFF),
+                                     title="music status")
+        if ctx.voice_client is None:
+            return await ctx.send(embed=status_embed.add_field(
+                name="status", value="not playing"))
+
+        playing = ctx.voice_client.is_playing()
+        paused = ctx.voice_client.is_paused()
+        if playing or paused:
+            status_embed.add_field(
+                name=f"currently {'playing' if playing else 'paused'} track",
+                value=mk_title_string(self.playing[0]),
+                inline=False,
+            ).add_field(
+                name="volume", value=f"{self.volume_level}%", inline=True
+            ).add_field(
+                name="track location",
+                value=
+                f"{time.strftime('%H:%M:%S', time.gmtime(self.compute_curr_time(paused)))}/"
+                f"{mk_duration_string([self.playing[0]])}",
+                inline=True).add_field(
+                    name="looping track",
+                    value="yes" if self.looping_track else "no",
+                    inline=True).add_field(
+                        name="playback speed",
+                        value=str(self.speed_val),
+                        inline=True).add_field(
+                            name="loop length" if self.looping_queue else
+                            "remaining queue length",
+                            value=f"{mk_duration_string(self.total_queue())}"
+                            f" ({self.total_len()} tracks)",
+                            inline=True).add_field(name="track submitter",
+                                                   value=str(self.playing[1]),
+                                                   inline=True)
+        else:
+            status_embed.add_field(
+                name="status",
+                value=
+                "not playing but connected [ERROR STATE SHOULD NEVER OCCUR]",
+            )
+        await ctx.send(embed=status_embed)
+
+    @commands.command(aliases=["rs", "rt"])
+    async def remove_track(self, ctx, track_index: int):
+        """Remove a track from the track queue by index"""
+
+        ltup = self.from_total(track_index)
+        if ltup is None:
+            return await ctx.send("removal index out of bounds.")
+        track_list, del_index = ltup
+
+        removed = mk_change_embed(track_list[del_index][0],
+                                  [track_list[del_index][0]],
+                                  f"removed track at index {track_index}",
+                                  f"removed by: {ctx.author}")
+        del track_list[del_index]
+        await ctx.send(embed=removed)
+
+    @commands.command(aliases=["iqs", "iqt"])
+    @check_banned
+    async def insert_track(self, ctx, track_index: int, *, url: str):
```

Comment:
right now this doesnt work if theres no song or just one song but i think it should